ec/README.md

```diff
@@ -6,18 +6,158 @@
-* [`CurveCycle`](https://github.com/arkworks-rs/algebra/blob/master/ec/src/lib.rs#L319) - Trait representing a cycle of elliptic curves.
-* [`PairingFriendlyCycle`](https://github.com/arkworks-rs/algebra/blob/master/ec/src/lib.rs#L331) - Trait representing a cycle of pairing friendly elliptic curves.
+### The `Group` trait
 
-The elliptic curve models implemented are:
+Many cryptographic protocols use as core building-blocks prime-order groups. The [`Group`](https://github.com/arkworks-rs/algebra/blob/master/ec/src/lib.rs) trait is an abstraction that represents elements of such abelian prime-order groups. It provides methods for performing common operations on group elements:
+
+```rust
+use ark_ec::Group;
+use ark_ff::{PrimeField, Field};
+// We'll use the BLS12-381 G1 curve for this example.
+// This group has a prime order `r`, and is associated with a prime field `Fr`.
+use ark_test_curves::bls12_381::{G1Projective as G, G1Affine as GAffine, Fr as ScalarField};
+use ark_std::{Zero, UniformRand, ops::Mul};
+
+let mut rng = ark_std::test_rng();
+// Let's sample uniformly random field elements:
+let a = G::rand(&mut rng);
+let b = G::rand(&mut rng);
+
+// We can add elements, ...
+let c = a + b;
+// ... subtract them, ...
+let d = a - b;
+// ... and double them.
+assert_eq!(c + d, a.double());
+// We can also negate elements, ...
+let e = -a;
+// ... and check that negation satisfies the basic group law
+assert_eq!(e + a, G::zero());
+
+// We can also multiply group elements by elements of the corresponding scalar field
+// (an act known as *scalar multiplication*)
+let scalar = ScalarField::rand(&mut rng);
+let e = c.mul(scalar);
+let f = e.mul(scalar.inverse().unwrap());
+assert_eq!(f, c);
+```
+
+## Scalar multiplication
+
+While the `Group` trait already produces scalar multiplication routines, in many cases one can take advantage of
+the group structure to perform scalar multiplication more efficiently. To allow such specialization, `ark-ec` provides
+the `ScalarMul` and `VariableBaseMSM` traits. The latter trait computes an "inner product" between a vector of scalars `s` and a vector of group elements `g`. That is, it computes `s.iter().zip(g).map(|(s, g)| g * s).sum()`.
+
+```rust
+use ark_ec::{Group, VariableBaseMSM};
+use ark_ff::{PrimeField, Field};
+// We'll use the BLS12-381 G1 curve for this example.
+// This group has a prime order `r`, and is associated with a prime field `Fr`.
+use ark_test_curves::bls12_381::{G1Projective as G, G1Affine as GAffine, Fr as ScalarField};
+use ark_std::{Zero, UniformRand};
+
+let mut rng = ark_std::test_rng();
+// Let's sample uniformly random field elements:
+let a = GAffine::rand(&mut rng);
+let b = GAffine::rand(&mut rng);
+
+let s1 = ScalarField::rand(&mut rng);
+let s2 = ScalarField::rand(&mut rng);
+let result = G::msm(&[a, b], &[s1, s2]);
```

Comment:
something that reads odd, why is it group MSM, but GAffine base points?

(I understand the reason, just not sure if we want to note anything about this in README, or for simplicity make a,b in G)